Security/Member.php

```diff
@@ -812,10 +812,16 @@ public function getValidator() {
 	 * @return Member
 	 */
 	public static function currentUser() {
+		static $cache = array();
+		
 		$id = Member::currentUserID();
 
 		if($id) {
-			return Member::get()->byID($id);
```

Comment:
replace this with `DataObject::get_by_id()`. It will provide a cache for you.

Then you don't need the caching code.